app/models/post.rb

```diff
@@ -1,3 +1,15 @@
 class Post < ApplicationRecord
-  belongs_to :user
+  belongs_to :author, class_name: 'User'
+  has_many :likes, foreign_key: :post_id
+  has_many :comments, foreign_key: :post_id
+
+  before_save :update_post_counter
```

Comment:
- [ ] [Optional] Even though this works as well, it would be nice to use the `after_save` callback for the sake of safety
```suggestion
  after_save :update_post_counter
```